tests/pyoozie/test_builder.py

```diff
@@ -134,6 +134,29 @@ def test_workflow_builder():
     actual_xml = builder.build()
     assert xml_to_dict_unordered(expected) == xml_to_dict_unordered(actual_xml)
 
+    # Does it throw an exception on a bad name?
+    with pytest.raises(AssertionError) as assertion_info:
+        WorkflowBuilder(
+            name='descriptive-name'
+        ).add_action(
+            name='Name with invalid characters',
+            action=Shell(exec_command='echo "test"'),
+            action_on_error=Email(to='person@example.com', subject='Error', body='A bad thing happened'),
+            kill_on_error='Failure message',
+        )
+    assert str(assertion_info.value) == \
+        "Identifier must match ^[a-zA-Z_][\\-_a-zA-Z0-9]{0,38}$, 'Name with invalid characters' does not"
```

Comment:
Upon closer inspection, Oozie may have intended to limit identifiers to 39 characters, but because to the `*` they do not. My regex follows the spirit of their regex, but not what it actually restricts.

Should we use their regex with no length limit? Or my more limited regex?